### PR TITLE
add OSB API Header version flag

### LINKED ISSUE #810: Add support for setting the OSB API Header version via config option
Mainly so that when we move to v2.12 if people need to roll back to v2.11 they need a way to override our default.

### PR DESCRIPTION
**Pull request by MHBauer**
- use version 2.12 explicitly for now instead of 'latest'

resolves #810 

/cc @duglin @pmorie @arschles

### CODE REVIEW

**Comment by pmorie** [comment]
I think there should be an API field for this and it should be set by an admission controller, per my comment here: https://github.com/kubernetes-incubator/service-catalog/issues/982#issuecomment-311695890

I think the best experience for users is where the broker resource is the single source of information about the API version in use.  It looks to me like there is a configuration mechanism for admission controllers that we can leverage - perhaps @derekwaynecarr can weigh in on that.  I hope that we would be able to retain the CLI wiring you have put in here to wire the default the admission controller uses.

File: cmd/controller-manager/app/options/options.go [hunk 1]
```diff
@@ -84,7 +88,9 @@ func (s *ControllerManagerServer) AddFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&s.ServiceCatalogKubeconfigPath, "service-catalog-kubeconfig", "", "Path to service-catalog kubeconfig")
 	fs.DurationVar(&s.ResyncInterval, "resync-interval", s.ResyncInterval, "The interval on which the controller will resync its informers")
 	fs.DurationVar(&s.BrokerRelistInterval, "broker-relist-interval", s.BrokerRelistInterval, "The interval on which a broker's catalog is relisted after the broker becomes ready")
-	fs.BoolVar(&s.OSBAPIContextProfile, "enable-osb-api-context-profile", s.OSBAPIContextProfile, "Whether or not to send the proposed optional OpenServiceBroker API Context Profile field")
+	fs.BoolVar(&s.OSBAPIContextProfile, "enable-osb-api-context-profile", s.OSBAPIContextProfile, "This does nothing.")
+	fs.MarkHidden("enable-osb-api-context-profile")
```
Comment (duglin) [inline cmd/controller-manager/app/options/options.go]: Why do we have a hidden flag that "does nothing" ?

**Comment by duglin** [comment]
@pmorie I think @MHBauer was doing just the global flag in this PR. I think he thought that someone else (you?) might have been working on the per-broker version thingy.  If not, then yea we might as well do both in one PR.

**Comment by pmorie** [comment]
I think the only global flag should be one that configures an admission controller that sets the value of the version field on the broker if it isn't already set.

File: cmd/controller-manager/app/options/options.go (hunk 1 above)
Comment (MHBauer) [inline cmd/controller-manager/app/options/options.go]: https://github.com/kubernetes-incubator/service-catalog/issues/982#issuecomment-311145030

**Comment by MHBauer** [comment]
Was going to do this and the api change separately. It's fairly separate work.

**Comment by pmorie** [comment]
@MHBauer Do we agree that the behavior when you're done (assuming you want to finish the task of adding the API surface per brokers) that the API version for a broker should always be explicit in the spec of that broker?  If so, I'm okay adding the knob for now.  Eventually I think such a knob should be logically on the API server, driving an admission controller that sets the field on the broker when it is created.  I am not sure if there is a good way to wire CLI options into admission controllers - @derekwaynecarr do you have a quick confirm or deny on that?

**Comment by duglin** [comment]
>  that the API version for a broker should always be explicit in the spec of that broker

If you mean its a required property on the spec from the client then I think it should be optional and we'd default it when we send OSB API messages.

If you mean its explicit when stored in etcd then I'm not sure since as we upgrade svc-cat from one version of the OSB API to another I would think that the OSB API version level (for existing brokers in the catalog) would automatically change with it, unless someone went out of their way to say "this broker must user version X of the spec".  Otherwise, as people upgrade svc-cat they would need to manually change all brokers to a new OSB API version number, which would be kind of annoying.

**Comment by MHBauer** [comment]
@pmorie @duglin let's move this discussion back over to #982.

**Comment by pmorie** [comment]
I'm okay with this PR, thanks for the patch.  LGTM

**Pull request merged**